include/aws/http/connection.h

```diff
@@ -478,6 +478,41 @@ int aws_http2_connection_ping(
     aws_http2_on_ping_complete_fn *on_completed,
     void *user_data);
 
+/**
+ * Asynchronously request to send a GOAWAY frame. (HTTP/2 only).
+ *
+ * @param http2_connection HTTP/2 connection.
+ * @param http2_error The HTTP/2 error code (RFC-7540 section 7) sent by peer.
+ *      `enum aws_http2_error_code` lists official codes.
+ * @param allow_more_streams If set, the graceful shutdown warning will be sent, new streams from peer will be allowed
+ *      until the connection shuts down. Note: If set, the sent http2_error will be set to AWS_HTTP2_ERR_NO_ERROR
```

Comment:
Alternative: AWS_INVALID_ARGUMENT for allow_more_streams=true && http2_error!=AWS_HTTP2_ERR_NO_ERROR